util/testutil/addcat/addcat.go

```diff
@@ -0,0 +1,39 @@
+package testutil
+
+import (
+	"bytes"
+	"strings"
+
+	core "github.com/ipfs/go-ipfs/core"
+	coreunix "github.com/ipfs/go-ipfs/core/coreunix"
+	unixfs "github.com/ipfs/go-ipfs/shell/unixfs"
+	errors "github.com/ipfs/go-ipfs/util/debugerror"
+)
+
+func AddCat(adder *core.IpfsNode, catter *core.IpfsNode, data string) error {
```

Comment:
I think we will want to use a `[]byte` here simply because so many other potential callers of this will generate data in `[]byte` form, and turning that into a `string` type requires a full copy of the entire data. For larger `AddCat` tests, this may be copying hundreds of MBs of data, not something i'm a fan of